# Individual Planning Report  
#### github link: https://github.com/Yifan669/Project_individual_plan.git

## load package

In [1]:
library(tidyverse)
library(tidymodels)

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors
── Attaching packages ────────────────────────────────────── tidymodels 1.1.1 ──

✔ broom        1.0.6     ✔ rsample      1.2.1
✔ dials        1.3.0     ✔ tune         1.1.2
✔ infer        1.0.7     ✔ workflows    1.1.4
✔ modeldata    1.4.0     ✔ workflowsets 1.0.1
✔ parsnip      1.2.1     ✔ yardstick    1.3.1
✔ recipes      1.1.0     

── Conflicts ───────────────────────────────────────── tidymodels_conflicts() ──
✖ scales::discard() masks purrr::discard()
✖ dplyr::filt

## Import data(Players)

In [2]:
url = 'https://drive.google.com/uc?export=download&id=1CwjG1G1zX1CAuvYRNh7vd07ETVWACEiA'
players <- read_csv(url)
head(players)

Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


experience subscribe
1 Pro         TRUE    
2 Veteran     TRUE    
3 Veteran    FALSE    
4 Amateur     TRUE    
5 Regular     TRUE    
6 Amateur     TRUE    
  hashedEmail                                                      played_hours
1 f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d 30.3        
2 f3c813577c458ba0dfef80996f8f32c93b6e8af1fa939732842f2312358a88e9  3.8        
3 b674dd7ee0d24096d1c019615ce4d12b20fcbff12d79d3c5a9d2118eb7ccbb28  0.0        
4 23fe711e0e3b77f1da7aa221ab1192afe21648d47d2b4fa7a5a659ff443a0eb5  0.7        
5 7dc01f10bf20671ecfccdac23812b1b415acd42c2147cb0af4d48fcce2420f3e  0.1        
6 f58aad5996a435f16b0284a3b267f973f9af99e7a89bee0430055a44fa92f977  0.0        
  name      gender Age
1 Morgan    Male    9 
2 Christian Male   17 
3 Blake     Male   17 
4 Flora     Female 21 
5 Kylie     Male   21 
6 Adrian    Female 17

## Summarize Data(Players)

In [3]:
played_hours_summary <- players |>
  summarize(played_hours_mean = mean(played_hours, na.rm = TRUE),
            played_hours_min  = min(played_hours,  na.rm = TRUE),
            played_hours_max  = max(played_hours,  na.rm = TRUE))

age_summary <- players |>
  summarize(age_mean = mean(Age, na.rm = TRUE),
            age_min  = min(Age,  na.rm = TRUE),
            age_max  = max(Age,  na.rm = TRUE))

played_hours_summary
age_summary

played_hours_mean played_hours_min played_hours_max
1 5.845918          0                223.1

age_mean age_min age_max
1 21.13918 9       58

In [4]:
experience_categories <- players |>
     distinct(experience)
experience_categories
gender_categories <- players |>
     distinct(gender)
gender_categories

experience
1 Pro       
2 Veteran   
3 Amateur   
4 Regular   
5 Beginner

gender           
1 Male             
2 Female           
3 Non-binary       
4 Prefer not to say
5 Agender          
6 Two-Spirited     
7 Other

## Data Description
Players dataset overview:
+ number of observations: 196
+ number of variables: 7  

The variables are:
+ `experience` is a character variable that records player's experience level in the game, such as `Pro`,`Veteran`, `Amateur`, `Regular`, and `Beginner`.
+ `subscribe` is a logical variable that indicates whether the players subscribed to a game-related newsletter, with `TRUE` meaning subscribed and `FALSE` meaning not subscribed.
+ `hashedEmail` is a character variable that records the uniquely hashed email addresses of players.
+ `played_hours` is a double variable that records the total number of hours each player has played on the MineCraft server. The mean `played_hours` is 5.85 hours, with a minimum of 0 and a maximum of 223.10.
+ `name` is a character variable that records the player's in-game name.
+ `gender` is a character variable that records the player’s reported gender, such as `Male`, `Female`, `Non-binary`, `Prefer not to say`, `Agender`, `Two-Spirited`, and `Other`.
+ `Age` is a double variable that records the player's age in years. The mean `Age` is 21.14, with a minimum of 9 and a maximum of 58.